type-overrides/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+///
+/// This lib contains both standard imports and imports shuttle.
+/// Shuttle is a Rust crate that facilitates multithreaded testing. It has its own scheduler
+/// and can efficiently detect bugs in concurrent code. The downside is that we need to replace
+/// all imports by those from Shuttle.
+///
+/// Instead of importing from std, rand, and so on, import the following from solana-type-override,
+/// and include the 'shuttle-test' feature in your crate to use shuttle.
+pub mod executor {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use futures::executor::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::future::*;
+}
+
+pub mod hint {
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::hint::*;
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use std::hint::*;
+}
+
+pub mod lazy_static {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use lazy_static::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::lazy_static::*;
+}
+
+pub mod rand {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use rand::{thread_rng, Rng, RngCore};
```

Comment:
why is this not `*` like the others?

Reply:
Shuttle only implements these three items from the rand crate, but we use more than those throught the code (I think I saw one of these cases in the rBPF crate).

When we import other items, we need to to do directly from rand. They work seamlessly with Shuttle.

Reply:
This is not great, since if shuttle starts implementing other types then we'll probably miss them. We could probably open a PR against shuttle to export type aliases for the original types in the rand prelude, or do it in our `rand` module.